go-controller/pkg/ovn/base_secondary_layer2_network_controller.go

```diff
@@ -247,7 +247,8 @@ func (oc *BaseSecondaryLayer2NetworkController) cleanup(topotype, netName string
 		return fmt.Errorf("failed to get ops for deleting switches of network %s: %v", netName, err)
 	}
 
-	ops, err = cleanupPolicyLogicalEntities(oc.nbClient, ops, netName)
+	controllerName := netName + "-network-controller"
```

Comment:
Would it make sense to have an utility function around building the controller name from the network name?

Reply:
You should be able to do so. Even when we build the dummy controllers...

https://github.com/ovn-org/ovn-kubernetes/blob/41fe592154e3e308b011a06422179bad6f356974/go-controller/pkg/network-controller-manager/network_controller_manager.go#L76

Reply:
So it actually uses controller to cleanup resources for other stale controllers/networks https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/network-controller-manager/network_controller_manager.go#L163
So we need to use netName here. There are 3 places where controller name is generated based on netName: https://github.com/search?q=repo%3Aovn-org%2Fovn-kubernetes+netInfo.GetNetworkName%28%29+%2B+%22-network-controller%22&type=code, I can probably just replace all of them with the new function call?

Reply:
Yeah, that looks good. 

 Why can't you use `oc.controllerName` here?

Reply:
I think dummy controllers don't have controller name, it is of type https://github.com/ovn-org/ovn-kubernetes/blob/76203a69c27ebc48762b7131f546cab15a57ed2c/go-controller/pkg/network-attach-def-controller/network_attach_def_controller.go#L53
which doesn't have controllerName

Reply:
Ah I think I see what the problem is. So I see 2 options:
1. I use given netName from the function argument
2. I change function signature to make it explicit that I am not using netName
The interface is defined here https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/network-attach-def-controller/network_attach_def_controller.go#L61 and it used by controllers like here https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/clustermanager/secondary_network_cluster_manager.go#L183, which are created here https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/clustermanager/secondary_network_cluster_manager.go#L172, and this controller doesn't seem to have controllerName, which makes me think I can't change the interface. But maybe I am also missing something here

Reply:
after some more discussion, we decided to try to change interface to not pass netNamt, as it doesn't seem necessary. This change will be done in a different PR, as it is irrelevant in the current context

Reply:
https://github.com/ovn-org/ovn-kubernetes/pull/4258